src/containers/SelectableTable.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import keyBy from 'lodash/keyBy';
+import mapValues from 'lodash/mapValues';
+import pickBy from 'lodash/pickBy';
+import keys from 'lodash/keys';
+import has from 'lodash/has';
+import filter from 'lodash/filter';
+
+import { Table } from '@patternfly/react-table';
+
+export default class SelectableTable extends React.Component {
+
+    state = {
+        selected: {}
+    }
+
+    onSelectAll (isSelected) {
+        if (!isSelected) {
+            return {};
+        }
+
+        return mapValues(keyBy(filter(this.props.rows, row => has(row, 'id')), r => r.id), () => true);
+    }
+
+    onSelectOne (state, isSelected, index) {
+        const row = this.props.rows[index];
+
+        if (!has(row, 'id')) {
+            return state;
+        }
+
+        if (!isSelected) {
+            return pickBy(state, (value, key) => key !== row.id);
+        }
+
+        return {
+            ...state,
```

Comment:
Do not spread state here, but `state.selected` to return already selected items.
```JS
return {
    ...state.selected,
    [row.id]: true
}
```